src/vcpkg/dependencies.cpp

```diff
@@ -380,34 +380,22 @@ namespace vcpkg
         };
     }
 
-    static std::string to_output_string(RequestType request_type,
-                                        const ZStringView s,
-                                        const BuildPackageOptions& options,
-                                        const SourceControlFileAndLocation* scfl,
-                                        const InstalledPackageView* ipv,
-                                        const Path& builtin_ports_dir)
+    static std::string to_output_string(const InstallPlanAction& action, const Path& builtin_ports_dir)
     {
-        std::string ret;
-        switch (request_type)
-        {
-            case RequestType::AUTO_SELECTED: Strings::append(ret, "  * "); break;
-            case RequestType::USER_REQUESTED: Strings::append(ret, "    "); break;
-            default: Checks::unreachable(VCPKG_LINE_INFO);
-        }
-        Strings::append(ret, s);
-        if (scfl)
+        std::string ret = to_output_string(action.request_type, action.displayname());
+        if (auto scfl = action.source_control_file_and_location.get())
         {
             Strings::append(ret, " -> ", scfl->to_version());
         }
-        else if (ipv)
+        else if (auto ipv = action.installed_package.get())
         {
             Strings::append(ret, " -> ", Version{ipv->core->package.version, ipv->core->package.port_version});
         }
-        if (options.use_head_version == UseHeadVersion::YES)
+        if (action.build_options.use_head_version == UseHeadVersion::YES)
         {
             Strings::append(ret, " (+HEAD)");
         }
-        if (scfl)
+        if (auto scfl = action.source_control_file_and_location.get())
         {
             if (!builtin_ports_dir.empty() &&
                 !Strings::case_insensitive_ascii_starts_with(scfl->source_location, builtin_ports_dir))
```

Comment:
It seems like InstallPlanAction should know this kind of prognosis information.... no change requested but if that were true a lot of complexity trying to pass this value around would go away.

Reply:
Agreed, SCFL should probably carry a bit around to indicate "unexpected location".